torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1015,6 +942,8 @@ void LoopNest::splitWithTail(
 
     Stmt* body_tail =
         Substitute(Stmt::clone(f->body()), {{f->var(), combined_index2}});
+    ReductionInitCleaner cleaner;
+    body_tail = cleaner.clean(body_tail);
```

Comment:
This looks perfectly fine. But from a design perspective, should what ReductionInitCleaner does logically be part of the clone?